CMakeLists.txt

```diff
@@ -258,9 +258,6 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
 set(CMAKE_C_STANDARD 99)
 set(CMAKE_C_STANDARD_REQUIRED ON)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
-set(CMAKE_RUNTIME_OUTPUT_DIRECTORY ${PX4_BINARY_DIR})
```

Comment:
@jgoppert shouldn't this be made conditional based on the type of build (cmake vs catkin), and set or not set based on it instead of just deleting? Does this still work with normal cmake build?

Reply:
From CI, it looks like it doesn't.

Reply:
Well all of these changes are actually redundant, the CMAKE_RUNTIME_OUTPUT_DIR=PX4_BINARY_DIR, which is set to CMAKE_BINARY_DIR, what breaks the devel path build for ROS is when the default is changed to ${PX4_BINARY_DIR}/bin in the lower directories, which basically clips the path so that it is shorter. I think this was probably done because people didn't like digging through all the subdirectories to find the px4 sitl binary.

Reply:
So.. yes.. it does change where subdirectories put built targets, which confuses the default make commands for px4. For sitl, CMAKE_RUNTIME_OUTPUT_DIR is set again, so that is redundant there.

Reply:
The quickest hack to to wrap all of these changes in a catkin conditional like you said @TSC21.